src/metabase/public_settings.clj

```diff
@@ -809,46 +809,78 @@ See [fonts](../configuring-metabase/fonts.md).")
                     ;; frontend should set this value to `true` after the modal has been shown once
                     v))))
 
-(defsetting uploads-enabled
-  (deferred-tru "Whether or not uploads are enabled")
-  :visibility :authenticated
-  :export?    true
-  :type       :boolean
-  :audit      :getter
-  :default    false)
-
 (defn- not-handling-api-request?
   []
   (nil? @api/*current-user*))
 
-(defn set-uploads-database-id!
-  "Sets the :uploads-database-id setting, with an appropriate permission check."
-  [new-id]
-  (if (or (not-handling-api-request?)
-          (mi/can-write? :model/Database new-id))
-    (setting/set-value-of-type! :integer :uploads-database-id new-id)
-    (api/throw-403)))
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Deprecated uploads settings begin
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; These settings were removed in v50 and will be removed in a future release. They are kept here for backwards compatibility
+;; to avoid breaking existing installations that may have set these values with environment variables, or via the config file.
+
+(defsetting uploads-enabled
```

Comment:
I do not fully understand why we still need these here. What's wrong if we remove them now?

Reply:
1. Without these, if you start MB with these settings set via the config file, Metabase will fail to launch. 
2. Without these, if you start MB with these settings set via env vars, it might be confusing because you wouldn't see why they weren't working anymore, even though you expected them to work (although that's much less important than 1).

Reply:
This was agreed with Vamsi [here](https://metaboat.slack.com/archives/C010L1Z4F9S/p1716515618806199?thread_ts=1716333848.276729&cid=C010L1Z4F9S)